modules/basics/src/main/java/com/opengamma/strata/basics/currency/Money.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.currency;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+
+import org.joda.beans.JodaBeanUtils;
+import org.joda.convert.ToString;
+
+import com.google.common.collect.ComparisonChain;
+import com.google.common.math.DoubleMath;
+
+/**
+ * An amount of a currency, rounded to match the currency specifications.
+ * <p>
+ * This class is similar to {@link CurrencyAmount}, but only exposes the rounded amounts.
+ * The rounding is done using {@link BigDecimal}, as BigDecimal.ROUND_HALF_EVEN. Given this operation,
+ * it should be assumed that the numbers are an approximation, and not an exact figure.
+ * <p>
+ * This class is immutable and thread-safe.
+ */
+public class Money implements FxConvertible<Money>, Comparable<Money>, Serializable {
+
+  /** Serialization version. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * The currency.
+   * For example, in the value 'GBP 12.34' the currency is 'GBP'.
+   */
+  private final Currency currency;
+
+  /**
+   * The amount of the currency.
+   * For example, in the value 'GBP 12.34' the amount is 12.34.
+   */
+  private final BigDecimal amount;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@code Money} for the specified currency and amount.
+   *
+   * @param currencyAmount the instance of {@link CurrencyAmount} wrapping the currency and amount.
+   * @return the currency amount
+   */
+  public static Money of(CurrencyAmount currencyAmount) {
+    Currency currency = currencyAmount.getCurrency();
+    BigDecimal roundedAmount = new BigDecimal(currencyAmount.getAmount())
+        .setScale(currency.getMinorUnitDigits(), BigDecimal.ROUND_HALF_EVEN);
+    return new Money(currency, roundedAmount);
+  }
+
+  /**
+   * Obtains an instance of {@code Money} for the specified currency and amount.
+   *
+   * @param currency the currency the amount is in
+   * @param amount the amount of the currency to represent
+   * @return the currency amount
+   */
+  public static Money of(Currency currency, double amount) {
+    BigDecimal roundedAmount = new BigDecimal(amount)
+        .setScale(currency.getMinorUnitDigits(), BigDecimal.ROUND_HALF_EVEN);
+    return new Money(currency, roundedAmount);
+  }
+
```

Comment:
It's surprising there isn't a factory method taking a `Currency` and `BigDecimal`